lib/tests/streamlit/runtime/pages_manager_test.py

```diff
@@ -0,0 +1,202 @@
+
+        assert self.pages_manager.get_active_script_hash() == main_script_hash
+
+    def test_get_page_script_valid_hash(self):
+        """Ensure the page script is provided with valid page hash specified"""
+
+        self.pages_manager.set_script_intent("page_hash", "")
+        self.pages_manager.set_pages({"page_hash": {"page_script_hash": "page_hash"}})
+
+        page_script = self.pages_manager.get_page_script(
+            self.pages_manager.main_script_hash
+        )
+        assert page_script["page_script_hash"] == "page_hash"
+
+    def test_get_page_script_invalid_hash(self):
+        """Ensure the page script is provided with invalid page hash specified"""
+
+        self.pages_manager.set_script_intent("bad_hash", "")
+        self.pages_manager.set_pages({"page_hash": {"page_script_hash": "page_hash"}})
+
+        page_script = self.pages_manager.get_page_script(
+            self.pages_manager.main_script_hash
+        )
+        assert page_script is None
+
+    def test_get_page_script_valid_name(self):
+        """Ensure the page script is provided with valid page name specified"""
+
+        self.pages_manager.set_script_intent("", "page_name")
+        self.pages_manager.set_pages(
+            {
+                "page_hash": {
+                    "page_script_hash": "page_hash",
+                    "url_pathname": "page_name",
+                }
+            }
+        )
+
+        page_script = self.pages_manager.get_page_script(
+            self.pages_manager.main_script_hash
+        )
+        assert page_script["page_script_hash"] == "page_hash"
+
+    def test_get_page_script_invalid_name(self):
+        """Ensure the page script is not provided with invalid page name specified"""
+
+        self.pages_manager.set_script_intent("", "foo")
+        self.pages_manager.set_pages(
+            {
+                "page_hash": {
+                    "page_script_hash": "page_hash",
+                    "url_pathname": "page_name",
+                }
+            }
+        )
+
+        page_script = self.pages_manager.get_page_script(
+            self.pages_manager.main_script_hash
+        )
+        assert page_script is None
+
+    def test_get_initial_active_script(self):
+        """Test that the initial active script is correctly retrieved with the
+        main script path provided."""
+        page_info = self.pages_manager.get_initial_active_script("page_hash", "")
+
+        self.assertDictEqual(
+            page_info,
+            {"script_path": "main_script_path", "page_script_hash": "page_hash"},
+        )
+
+
+# NOTE: We write this test function using pytest conventions (as opposed to
+# using unittest.TestCase like in the rest of the codebase) because the tmpdir
+# pytest fixture is so useful for writing this test it's worth having the
+# slight inconsistency.
+@patch("streamlit.source_util._cached_pages", new=None)
+def test_get_initial_active_script_v1(tmpdir):
```

Comment:
nit: `tmpdir` seems to be a [py.path.local](https://py.readthedocs.io/en/latest/path.html#py-path-local-local-file-system-path) object according to the [pytest fixture docs](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture), in case you want to type it

Reply:
Hmmmm...I know this is silly, but py is not in our dependencies. I'm sure it would work if I import it (cause pytest must require it), but I don't want to add test dependency and have that break our tests. Open to other ideas if you feel this is important.

Reply:
that makes 100% sense to me to not use any transient dependencies!